code/modules/mob/living/carbon/human/login.dm

```diff
@@ -1,5 +1,9 @@
 /mob/living/carbon/human/Login()
 	. = ..()
+
+	if(ai_controller)
+		QDEL_NULL(ai_controller)
```

Comment:
this is supposed to already be handled by the ai_controller datum itself since it registers to mob login and mob logout. if you want the datum to be deleted when a player logs in then delete it from on_sentience_gained()